### ISSUE TITLE
undefined symbol: opj_encoder_set_extra_options on Python 3.9 on arm (raspberry pi)

### ISSUE THREAD

**Issue by bellini666** [python-pillow/Pillow#7067]
Updated to the 9.5.0 and got this error when trying to `from PIL import Image`:

```
ImportError: /home/pi/nomad/.nomad_venv/lib/python3.9/site-packages/PIL/_imaging.cpython-39-arm-linux-gnueabihf.so: undefined symbol: opj_encoder_set_extra_options
```

Downgrading to `9.4.0` fixed the issue.

**Comment by nulano** [comment]
How did you install Pillow?

I'm guessing with `python -m pip install -U Pillow`, which used a piwheels wheel?
You probably need to update your system OpenJPEG version to use their wheel, from https://www.piwheels.org/project/pillow/:
```bash
sudo apt install libwebpdemux2 libopenjp2-7 libwebpmux3 liblcms2-2
```

**Comment by bellini666** [comment]
@nulano it is installed in a venv. Distro is Raspbian buster, all packages up-to-date.

```
(venv) pi@raspberrypi:/tmp $ sudo apt install libwebpdemux2 libopenjp2-7 libwebpmux3 liblcms2-2
Reading package lists... Done
Building dependency tree       
Reading state information... Done
liblcms2-2 is already the newest version (2.9-3).
libopenjp2-7 is already the newest version (2.3.0-2+deb10u2).
libwebpdemux2 is already the newest version (0.6.1-2+deb10u1).
libwebpmux3 is already the newest version (0.6.1-2+deb10u1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

**Comment by nulano** [comment]
I'm guessing you are on Buster?

From https://www.openjpeg.org/doxygen/openjpeg_8c.html#a8fec822dd650a82db457ff1aafbc9b2a, it looks like `opj_encoder_set_extra_options` was added in OpenJPEG 2.4.0.

However, from https://www.raspberryconnect.com/raspbian-packages/78-raspbian-libs, although OpenJPEG 2.4.0 is available on Bullseye, it seems only OpenJPEG 2.3.0 is available on Buster:
> libopenjp2-7Bullseye:(2.4.0-3) Buster:(2.3.0-2+deb10u2) JPEG 2000 image compression/decompression library [www](https://www.openjpeg.org/)

So it seems the unofficial wheel https://www.piwheels.org/simple/pillow/Pillow-9.5.0-cp39-cp39-linux_armv7l.whl is not compatible with Buster, only Bullseye.

You might want to open an issue at https://github.com/piwheels/packages instead.

---

Alternatively, you can try installing Pillow from source using the instructions at https://pillow.readthedocs.io/en/stable/installation.html#building-from-source

**Comment by bellini666** [comment]
@nulano that makes sense. Will report the issue there. Thanks!
